### DISCUSSION TITLE
Model Magic Sequence (Magic Series) problem using MIP

### DISCUSSION THREAD

**Discussion by dusantrtica** [coin-or/python-mip#310] [General]
Hello everyone,
I am trying to model Magic Sequence (or Magic Series) problem using MIP. Here is how I get started but it just doesn't work.
Could you please tell me what I am doing wrong here? Thanks

```
# Generates magic series:
# Example: for n = 5, one of the series is S = [2, 1, 2, 0, 0]
# Magic series S is an array such that S[i] is a number of occurences of i in S
# and its elements sum to n
n = 5
magic_series = Model('magic-series')

variables = [magic_series.add_var("", var_type=INTEGER, lb=0, ub=n-1) for i in range(n)]

for i in range(n):
    magic_series.add_constr(variables[i] == len([variable for variable in variables if variable == i]))

magic_series.add_constr(xsum(variables) == n)

magic_series.optimize()
```

**Comment by sebheger** [comment]
Hi @dusantrtica We do not offer support for modeling questions in general. If you can specify more in detail what doesn't work, we might if it is a bug.

**Reply by rschwarz** [reply]
A good place to ask modeling questions would be the [OR StackExchange](https://or.stackexchange.com/).

PS: You can't use condititionals such as `if variable == i` in a MIP formulation. The boolean expression will be evaluated by Python, but the value of `variable` is not yet known. Instead of using integer variables, try using binary values, one for each possible assignment.
